src/microbe_stage/Microbe.cs

```diff
@@ -1209,6 +1212,13 @@ private void HandleCompoundAbsorbing(float delta)
 
         cloudSystem.AbsorbCompounds(Translation, grabRadius, Compounds,
             TotalAbsorbedCompounds, delta, Membrane.Type.ResourceAbsorptionFactor);
+
+        if (!CheatManager.InfCompounds || !IsPlayerMicrobe)
```

Comment:
I think I'd prefer this if to be the other way around as this is now, this is pretty hard for me to wrap by head around.